File: src/LiWiMus.Core/Entities/Artist.cs
using System.ComponentModel.DataAnnotations;
using LiWiMus.Core.Constants;

namespace LiWiMus.Core.Entities;

public class Artist : BaseEntity
{
    public User? User { get; set; }

    [StringLength(50, MinimumLength = 5)]
    [RegularExpression(RegularExpressions.DisableTags)]
    public string Name { get; set; } = null!;

    [StringLength(500)]
    public string About { get; set; } = null!;

    public string PhotoPath { get; set; } = null!;

    public List<Track> Tracks { get; set; } = new();
    public List<LikedArtist> Subscribers { get; set; } = new();
    public List<Album> Albums { get; set; } = new();
}

File: src/LiWiMus.Web.Admin/client/src/artists/types/Artist.ts
import { Album } from "../../albums/types/Album";

export type Artist = {
  id: string;
  name: string;
  about: string;
  photoLocation: string;

  createdAt: string;
  modifiedAt: string;

  tracksCount: number;
  albumsCount: number;

  albums: Album[];
};
